### PR TITLE
Add Tab::loadData() API

### PR DESCRIPTION
**Pull request by svillar**
This API allows clients to load some data with a given mime type and encoding. This is typically used to load data:// URIs although it supports multiple combinations of mime type and encoding.

### CODE REVIEW

**Review by ns-mshin — APPROVED** [review]

File: wolvic/java/org/chromium/wolvic/Tab.java [hunk 1]
```diff
@@ -172,6 +179,40 @@ private int findBackForwardNavigationOffset(NavigationDirection direction) {
...
+        }
+
+        // If we are reloading the same url, then set transition type as reload.
+        if (params.getUrl() != null
+                && params.getUrl().equals(mWebContents.getLastCommittedUrl().getSpec())
+                && params.getTransitionType() == PageTransition.TYPED) {
+            params.setTransitionType(PageTransition.RELOAD);
+        }
+        params.setTransitionType(params.getTransitionType() | PageTransition.FROM_API);
+        params.setUrl(UrlFormatter.fixupUrl(params.getUrl()).getPossiblyInvalidSpec());
```
Comment (javifernandez) [inline wolvic/java/org/chromium/wolvic/Tab.java:205]: You said in the comments that this method loads the url without fixing up the url string, but here yu are updating the param's url.

**Review by javifernandez — APPROVED** [review]
Looks good, but check out the comments about the param's url fixup.

File: wolvic/java/org/chromium/wolvic/Tab.java (hunk 1 above)
Comment (svillar) [reply wolvic/java/org/chromium/wolvic/Tab.java:205]: Makes sense, I'll modify the comment accordingly

Comment (svillar) [reply wolvic/java/org/chromium/wolvic/Tab.java:205]: Finally removed it since it's a private method after all.

**Pull request merged**